daemon/tests/harness/mocks/monitor.rs

```diff
@@ -1,29 +1,53 @@
+use std::sync::Arc;
+
 use daemon::{monitor, oracle};
+use mockall::*;
+use tokio::sync::Mutex;
 use xtra_productivity::xtra_productivity;
 
-/// Test Stub simulating the Monitor actor
-pub struct Monitor;
-impl xtra::Actor for Monitor {}
+/// Test Stub simulating the Monitor actor.
+/// Serves as an entrypoint for injected mock handlers.
+pub struct MonitorActor {
+    pub mock: Arc<Mutex<dyn Monitor + Send>>,
+}
+
+impl xtra::Actor for MonitorActor {}
+impl Monitor for MonitorActor {}
 
 #[xtra_productivity(message_impl = false)]
-impl Monitor {
-    async fn handle(&mut self, _msg: monitor::Sync) {}
+impl MonitorActor {
+    async fn handle(&mut self, _msg: monitor::Sync) {
+        self.mock.lock().await.sync()
+    }
 
     async fn handle(&mut self, _msg: monitor::StartMonitoring) {
-        todo!("stub this if needed")
+        self.mock.lock().await.start_monitoring()
     }
 
     async fn handle(&mut self, _msg: monitor::CollaborativeSettlement) {
-        todo!("stub this if needed")
+        self.mock.lock().await.collaborative_settlement()
     }
 
     async fn handle(&mut self, _msg: oracle::Attestation) {
-        todo!("stub this if needed")
+        self.mock.lock().await.oracle_attestation()
     }
 }
 
-impl Default for Monitor {
-    fn default() -> Self {
-        Monitor
+#[automock]
+pub trait Monitor {
```

Comment:
I think it would be nice if we can route all the messages through by default. I think the mock should always reflect the signature of the `handle` function.

Reply:
I'll follow YAGNI here - let's add it when (or perhaps *if*) we need it. In this case, it would be the time someone wants to write a mock using it.

Reply:
ah, it was so simple, I just implemented it after all :-)